src/fmu/dataio/dataio.py

```diff
@@ -663,7 +618,7 @@ def _validate_workflow_key(self) -> None:
 
     def _validate_content_key(self) -> None:
         """Validate the given 'content' input."""
-        self._usecontent, self._content_specific = _check_content(self.content)
+        self._usecontent = _check_content(self.content)
```

Comment:
Maybe all of what `_check_content` is doing can just be moved into this method, and itself removed? Particularly as it's quite a bit smaller now the extra layer of indirection doesn't seem needed. What do you think?

Reply:
Removed these validation checks in dataio, and moved it inside the objdataprovider instead as discussed 👍